Added flPersistEvents flag to disable writing events to disk

Cliffstone opened the pull request:
**Description:**
This pull request introduces a flag, `-persist-events`, to moroz to addresses the issue of high directory creation and possible inode shortage in some cases along with increased memory usage.

**Changes:**
Added flPersistEvents field to SantaService struct in moroz/service.go.
Updated all references to SantaService to include the new flPersistEvents field.
Modified related functions and methods to accommodate the new flPersistEvents field.

**Impact:**
This change allows us to control whether events should be persisted, providing more flexibility for the user.  The default value is set to `false` meaning moroz will only print logs to stdout

**Testing:**
Tested locally and successfully stopped writing to disk by default. Setting -persist-events="true"  also worked like a charm
 ./moroz -debug=true -configs="./configs" -event-dir="./logs" -persist-events="true

Cliffstone committed:
first changes

kolbeinn commented on cmd/moroz/main.go:
```diff
@@ -42,14 +42,15 @@ The latest version of santa is available on the github repo page:
+		flAddr          = flag.String("http-addr", env.String("MOROZ_HTTP_ADDRESS", ":8080"), "http address ex: -http-addr=:8080")
+		flConfigs       = flag.String("configs", env.String("MOROZ_CONFIGS", "../../configs"), "path to config folder")
+		flEvents        = flag.String("event-dir", env.String("MOROZ_EVENT_DIR", "/tmp/santa_events"), "Path to root directory where events will be stored.")
+		flPersistEvents = flag.Bool("persist-events", env.Bool("MOROZ_WRITE_EVENTS", false), "Enable writing events to disk. Disabled by default.")
```
It might be easier to get this merged upstream if the default value preserves the current functionality of moroz

kolbeinn reviewed (commented):
Looks good. Just the one comment about the default value. Also, I would explain what you mean by "directory DoS" when you submit this upstream. Just be very explicit about what the problem is and how this PR solves it.

Cliffstone committed:
Update README.md

default value set to true

Cliffstone committed:
Update main.go

updated the flPersistEvents default value to true and fixed the description text

Cliffstone reviewed (commented):
Fixed
